### DISCUSSION TITLE
调用stable diffusion webui工具进行绘图失败“Failed to generate image”

### DISCUSSION THREAD

**Discussion by fengyingbanbo** [langgenius/dify#3647] [General]
stable diffusion webui 监听的端口是7862
![image](https://github.com/langgenius/dify/assets/24475157/afc4f3f7-d0bd-4c94-aa95-104b40d45b94)
 配置的是http://192.168.3.37:7862/，已经授权成功
![image](https://github.com/langgenius/dify/assets/24475157/aa81aabf-6e0c-4617-8876-dc68cee1c63f)

大语言模型用的llama3:8b，调用stable diffusion web ui工具返回{"stable_diffusion": "Failed to generate image"}
![image](https://github.com/langgenius/dify/assets/24475157/0d5312bd-f87c-40f1-aa6c-c4bf6e475fcb)

**Comment by Joycc** [comment]
Same error.
Log error: 
HTTP Request: POST http://172.21.100.64:7860/sdapi/v1/options "HTTP/1.1 200 OK"
HTTP Request: POST http://172.21.100.64:7860/sdapi/v1/txt2img "HTTP/1.1 404 Not Found"

**Reply by Joycc** [reply]
The default index value of webui is incorrect. Modify it directly in the dify/api/core/tools/provider/builtin/stablediffusion/tools/stable_diffusion.py
'sampler_index': 'DPM++ SDE Karras'
